### PR TITLE
Fixing DFA result cache

### PR DESCRIPTION
**Pull request by dotpaul**
Key DFA results by analysis context rather than operation visitor (which was missing its Equals() override, but let's just not use operation visitors as keys).

Also, since this is a pretty fundamental change to DFA, would appreciate any help dogfooding!

### CODE REVIEW

**Comment by dotpaul** [comment]
FYI @LLLXXXCCC @mavasani @timhermann @matt-desai

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs [hunk 1]
```diff
@@ -223,8 +223,6 @@ protected DataFlowOperationVisitor(TAnalysisContext analysisContext)
                 getInterproceduralCallStackForOwningSymbol: GetInterproceduralCallStackForOwningSymbol);
         }
 
-        public override int GetHashCode() => HashUtilities.Combine(GetType().GetHashCode(), DataFlowAnalysisContext.GetHashCode());
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs]: I believe this is overridden in other visitors (sub-types) too. Can you check?

Comment (dotpaul) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs]: Oops, yes, thanks.  Will also remove from others.

Any reason to think this change will introduce issues with DisposeAnalysis or ParameterValidationAnalysis?

Comment (mavasani) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs]: Yes, that is the primary thing I need to think on, as the operation visitors for those two are stateful. Let me get back to you early tomorrow morning.

**Comment by dotpaul** [comment]
Thanks @mavasani !

**Pull request merged**